src/discord_utils.py

```diff
@@ -31,6 +31,26 @@ async def try_set_pin(message: Message, pin_state: bool) -> None:
         print("Altering message pin state failed:", e)
 
 
+def attachment_local_filepath(message_id: int, attachment: Attachment) -> str:
+    """Build local filepath with provided Message ID & Attachment object for saving Attachments locally"""
```

Comment:
Looks like Jack and I never explained exactly *why* we formatted attachment's local filepaths in this manner (with the message ID and other bits of text). This docstring would be a handy place to do so.

```suggestion
    """Generate a unique filepath for a given attachment for the purposes of saving it to disk.
    
    Generates a filepath in the following format:
    
    <attachment_directory>/<Discord message ID>.<attachment ID>.<file extension>
    
    For example:
    
    /home/user/busty/625891304148303894.625891304081063986.mp3
    
    Filepaths for each song should be unique and the rest of the codebase assumes that they are.
    A Discord message can contain multiple attachments, including the attachment ID as well.
    
    Args:
    	message_id: The ID of the Discord message that included this attachment.
    	attachment: The attachment to build a filepath for.
    	
    Returns:
    	An absolute filepath to a file for this attachment in the configured attachment directory.
    """
```

Although now after having written this up, I'm wondering if we can just get away with using attachment IDs in order to get a unique filename. Ah well, an issue for another PR :)

Reply:
Yeah just using attachment ID is probably a better solution. Or better yet, use <server_id>/<attachment_id>.<file_ext> to move us closer to server-independence. But I agree we'll do this in a separate PR